### PR TITLE
Add route to copy permissions to role

### PR DESCRIPTION
**Pull request by vicpatel**
It copies all RBAC rules from one role to another, but also removes all existing rules from role.

Please take a look.

### CODE REVIEW

File: system/rest.yaml [hunk 1]
```diff
@@ -944,6 +944,20 @@ endpoints:
         type: rbac.RuleSet
         required: true
         title: List of permission rules to set
+  - name: copy
+    path: "/{roleID}/rules/{originRoleID}"
```
Comment (darh) [inline system/rest.yaml]: Path is a bit misleading... Give it another look please

File: system/rest/permissions.go [hunk 2]
```diff
@@ -61,3 +61,29 @@ func (ctrl Permissions) Update(ctx context.Context, r *request.PermissionsUpdate
 
 	return api.OK(), ctrl.ac.Grant(ctx, r.Rules...)
 }
+
+func (ctrl Permissions) Copy(ctx context.Context, r *request.PermissionsCopy) (interface{}, error) {
```
Comment (darh) [inline system/rest/permissions.go]: This is excellent -- but it's too much logic (that should be part of the service) in the controller.

File: system/service/user_actions.gen.go [hunk 3]
```diff
@@ -11,13 +11,12 @@ package service
...
 	"fmt"
-	"strings"
-	"time"
-
 	"github.com/cortezaproject/corteza-server/pkg/actionlog"
 	"github.com/cortezaproject/corteza-server/pkg/errors"
 	"github.com/cortezaproject/corteza-server/pkg/locale"
 	"github.com/cortezaproject/corteza-server/system/types"
+	"strings"
+	"time"
```
Comment (darh) [inline system/service/user_actions.gen.go:19]: this was probably a mistake?

File: system/service/auth_actions.gen.go [hunk 4]
```diff
@@ -11,13 +11,12 @@ package service
...
 	"fmt"
-	"strings"
-	"time"
-
 	"github.com/cortezaproject/corteza-server/pkg/actionlog"
 	"github.com/cortezaproject/corteza-server/pkg/errors"
 	"github.com/cortezaproject/corteza-server/pkg/locale"
 	"github.com/cortezaproject/corteza-server/system/types"
+	"strings"
+	"time"
```
Comment (darh) [inline system/service/auth_actions.gen.go:19]: this was probably a mistake?

**Review by darh — CHANGES_REQUESTED** [review]

File: system/service/auth_actions.gen.go (hunk 4 above)
Comment (vicpatel) [reply system/service/auth_actions.gen.go:19]: yes

File: pkg/rbac/service.go [hunk 5]
```diff
@@ -279,3 +279,31 @@ func (svc service) String() (out string) {
 
 	return
 }
+
+// CloneRulesByRoleID clone all rules of a Role S to a specific Role T
+//      returns set of role T rules by replacing it's existing rules with role S rules
+//
+// This function is auto-generated
```
Comment (darh) [inline pkg/rbac/service.go]: is it? :)

**Review by darh — CHANGES_REQUESTED** [review]
excellent. small (comment) fix, tests & you are done!

**Pull request merged**